### PR TITLE
chore: Rework deepcopy generators to not require gopath

### PR DESCRIPTION
**Pull request by justinsb**
We can use the upstream deepcopy implementation, eliminate our fork, and move this out of the section that requires GOPATH.

This is more maintainable and feels faster.

- chore: Rework deepcopy generators to not require gopath
- Remove not-generated deepcopy
- Update generated code

### CODE REVIEW

**Comment by cheftako** [comment]
Looks good but we need to fix the conflict.

**Comment by acpana** [comment]
Just for my own gratification, I rebased and integrated the changes locally and unfortunately the error in https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/1427 is still there 😢 . Probably have to dig deeper to understand why the new resources (from dcl 1.63 on) are causing this.

File: pkg/apis/core/v1alpha1/zz_generated.deepcopy.go
Comment (yuwenma) [inline pkg/apis/core/v1alpha1/zz_generated.deepcopy.go:1]: Just out of curiosity: how did you find out this file?

**Review by yuwenma — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**